File: frontend/src/services/ItemService.js
// ✅ This is your clean ItemService.js — place this in /services/ItemService.js

import axios from "axios";

const API = "http://localhost:8087/api/items";

export const getAllItems = () => axios.get(API);

export const createItem = (item) => axios.post(API, item);

export const updateItem = (id, item) => axios.put(`${API}/${id}`, item);

export const deleteItem = (id) => axios.delete(`${API}/${id}`);

export const importItems = (formData) =>
  axios.post(`${API}/items/import`, formData, {
    headers: {
      "Content-Type": "multipart/form-data",
    },
  });

// ✅ Note: Use `deleteItem` not `deleteItemById` in your import

File: backend/src/main/java/com/BillingApp/services/ItemService.java
package com.BillingApp.services;


import java.util.List;

import com.BillingApp.DTO.ItemDTO;
import com.BillingApp.model.Item;

public interface ItemService {
    Item createItem(Item item);

    ItemDTO addItem(ItemDTO dto);
    ItemDTO updateItem(Long id, ItemDTO dto);
    List<ItemDTO> getAllItems();
    void deleteItem(Long id);
    void saveAll(List<Item> items);

}
